### PR TITLE
treewide: simplify some dependency cycles

### PR DESCRIPTION
**Pull request by oxij**
Yet another followup to #39464.

### CODE REVIEW

**Review by matthewbauer — APPROVED** [review]

File: pkgs/development/python-modules/hypothesis/default.nix [hunk 1]
```diff
@@ -20,9 +20,9 @@ buildPythonPackage rec {
     sha256 = "063sn5m1966gvm3wrlxczdq4vw0r94h3nd9xpr94qxahpg2r4bpb";
   };
 
-  checkInputs = [ pytest pytest_xdist flaky mock ];
   propagatedBuildInputs = [ attrs coverage ] ++ lib.optional (!isPy3k) [ enum34 ];
 
+  checkInputs = [ pytest pytest_xdist flaky mock ];
```
Comment (FRidh) [inline pkgs/development/python-modules/hypothesis/default.nix]: unnecessary change

**Comment by oxij** [comment]
@FRidh

> unnecessary change

It's necessary for satisfying my OCD of grouping related things together.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -9528,10 +9528,8 @@ with pkgs;
   gtkmathview = callPackage ../development/libraries/gtkmathview { };
 
   glib = callPackage ../development/libraries/glib {
-  };
-  glib-tested = glib.override { # checked version separate to break cycles
-    doCheck = true;
-    libffi = libffi.override { doCheck = true; };
+    # dependency cycle
+    libffi = libffi.override { doCheck = false; };
```
Comment (orivej) [inline pkgs/top-level/all-packages.nix]: Is this override necessary? I can instantiate `glib` without it.

**Review by orivej — COMMENTED** [review]
/cc @vcunat @domenkozar
`glib-tested` was a part of `release.nix` until it was removed in c21f9365b805491351e500e53bff66d0e50af472, probably due to mysterious Nix failures on Hydra. Its build has not been failing since then.

**Comment by oxij** [comment]
> Is this override necessary? I can instantiate `glib` without it.

Oops, right, that's because `glib` has `doCheck` set to `true` by default in my branch and I have not upstreamed those patches yet.

I tried to split stuff into bite-sized pieces, but failed in this instance.

Then, the logical way to proceed is to close this PR and to postpone this discussion till I get those patches ready then.

**Comment by oxij** [comment]
On the second though, nothing prevents me from removing only that part from here too. Updated.
The glib things will go into a separate PR.

**Comment by oxij** [comment]
@GrahamcOfBorg python2Packages.pytest python2Packages.hypothesis

**Comment by LnL7** [comment]
FYI. This broke the darwin stdenv.

**Comment by matthewbauer** [comment]
Ok. Unless there is an alternative fix available we can just do ```doCheck ? !stdenv.isDarwin```, correct? That would restore old behavior until we can figure out what is going on.

**Comment by LnL7** [comment]
Yeah, I'm testing to see if there are any other problems.
